### ISSUE TITLE
OpenSSL defaults to http1.1

### ISSUE THREAD

**Issue by rforbes** [openssl-net/openssl-net#27]
I am setting up a new ssl tunnel using this library.  Everything works great except for the ALPN negotiation.  When I use firefox it sends h2-12, SPDY3, and HTTP1.1.  I am getting a negotiation with HTTP1.1.  If I use openssl with s_client and use the -alpn flag then the server will return h2-12 correctly.

What it looks like is openssl needs some way to specify what ALPN protocol should be negotiated.  Using s_server there is a flag -alpn we can specify this.  Is it possible to set something like that with this library?

**Comment by flaub** [comment]
Yes. I need to add an api to allow users to optionally enable features, like ALPN and SNI.

**Comment by rforbes** [comment]
any idea when this functionality could be added?

thanks!

**Comment by vtortola** [comment]
Same issue here :)  Calling with Google Chrome and getting HTTP 1.1
